Applications/Utils/MeshEdit/ResetPropertiesInPolygonalRegion.cpp

```diff
@@ -0,0 +1,215 @@
+
+// BaseLib
+#include "BaseLib/LogogSimpleFormatter.h"
+
+// FileIO
+#include "FileIO/readMeshFromFile.h"
+#include "FileIO/writeMeshToFile.h"
+#include "FileIO/XmlIO/Boost/BoostXmlGmlInterface.h"
+
+// GeoLib
+#include "GeoLib/GEOObjects.h"
+#include "GeoLib/Polygon.h"
+#include "GeoLib/AnalyticalGeometry.h"
+
+// MathLib
+#include "MathLib/Vector3.h"
+
+// MeshLib
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/Elements/Element.h"
+#include "MeshLib/MeshEditing/ElementValueModification.h"
+
+GeoLib::Polygon rotatePolygonToXY(GeoLib::Polygon const& polygon_in,
+	MathLib::Vector3 & plane_normal)
+{
+	// 1 copy all points
+	std::vector<GeoLib::Point*> *polygon_pnts(new std::vector<GeoLib::Point*>);
+	for (std::size_t k(0); k < polygon_in.getNumberOfPoints(); k++)
+		polygon_pnts->push_back (new GeoLib::Point (*(polygon_in.getPoint(k))));
+
+	// 2 rotate points
+	double d_polygon (0.0);
+	GeoLib::getNewellPlane (*polygon_pnts, plane_normal, d_polygon);
+	MathLib::DenseMatrix<double> rot_mat(3,3);
+	GeoLib::computeRotationMatrixToXY(plane_normal, rot_mat);
+	GeoLib::rotatePoints(rot_mat, *polygon_pnts);
+
+	// 3 set z coord to zero
+	std::for_each(polygon_pnts->begin(), polygon_pnts->end(),
+		[] (GeoLib::Point* p) { (*p)[2] = 0.0; }
+	);
+
+	// 4 create new polygon
+	GeoLib::Polyline rot_polyline(*polygon_pnts);
+	for (std::size_t k(0); k < polygon_in.getNumberOfPoints(); k++)
+		rot_polyline.addPoint (k);
+	rot_polyline.addPoint (0);
+	return GeoLib::Polygon(rot_polyline);
+}
+
+std::vector<bool> markNodesOutSideOfPolygon(
+	std::vector<MeshLib::Node*> const& nodes,
+	GeoLib::Polygon const& polygon)
+{
+	// *** rotate polygon to xy_plane
+	MathLib::Vector3 normal;
+	GeoLib::Polygon rot_polygon(rotatePolygonToXY(polygon, normal));
+
+	// *** rotate mesh nodes to xy-plane
+	// 1 copy all mesh nodes to GeoLib::Points
+	std::vector<GeoLib::Point*> rotated_nodes;
+	for (std::size_t j(0); j < nodes.size(); j++)
+		rotated_nodes.push_back(new GeoLib::Point(nodes[j]->getCoords()));
+	// 2 rotate the Points
+	MathLib::DenseMatrix<double> rot_mat(3,3);
+	GeoLib::computeRotationMatrixToXY(normal, rot_mat);
+	GeoLib::rotatePoints(rot_mat, rotated_nodes);
+	// 3 set z coord to zero
+	std::for_each(rotated_nodes.begin(), rotated_nodes.end(),
+		[] (GeoLib::Point* p) { (*p)[2] = 0.0; }
+	);
+
+	// *** mark rotated nodes
+	std::vector<bool> outside(rotated_nodes.size(), true);
+	for (std::size_t k(0); k<rotated_nodes.size(); k++) {
+		if (rot_polygon.isPntInPolygon(*(rotated_nodes[k]))) {
+			outside[k] = false;
+		}
+	}
+
+	for (std::size_t j(0); j < rotated_nodes.size(); j++)
+		delete rotated_nodes[j];
+
+	std::vector<GeoLib::Point*> & rot_polygon_pnts(
+		const_cast<std::vector<GeoLib::Point*> &>(
+			rot_polygon.getPointsVec()
+		)
+	);
+	for (std::size_t k(0); k < rot_polygon_pnts.size(); k++)
+		delete rot_polygon_pnts[k];
+
+	return outside;
+}
+
+void resetProperty(MeshLib::Mesh &mesh, GeoLib::Polygon const& polygon,
```

Comment:
you should note somewhere that you only reset elements that are *completely* located within the polygon.